rules/nat.go

```diff
@@ -48,18 +48,28 @@ func (r *DefaultRuleRenderer) MakeNatOutgoingRule(protocol string, action iptabl
 func (r *DefaultRuleRenderer) NATOutgoingChain(natOutgoingActive bool, ipVersion uint8) *iptables.Chain {
 	var rules []iptables.Rule
 	if natOutgoingActive {
+		var defaultSnatRule iptables.Action = iptables.MasqAction{}
+		if r.Config.NATOutgoingAddress != nil {
```

Comment:
Can this also check if the NATOutgoingAddress exists on an interface and fallback to masquerading if not? My use case is for floating/virtual IPs that will only exist on one of the calico nodes. But even without using a VIP it is good error checking to validate the IP exists.

Reply:
It seems like the correct place/time to check that would be on startup/reconfigure, but not sure if the correct behavior would be to error out or to warn and fallback. Happy to defer to the Calico folks on what it should be and implement it though.

Reply:
@steven-sheehy It doesn't feel right to me to add that feature into this PR.  I'd rather merge this PR first, then that can be added as a delta if it really needs a Felix code change.

Anyway, I wonder if this can be done (in a Kubernetes context) by combining an initContainer with the downward API?  Something like:
- use the downward API to populate an env var for an initContainer
- get the initContainer to check that that address really exists
- if so, also set it as FELIX_NATOUTGOINGADDRESS for the calico/node container.

Reply:
Ok, let's handle it in another PR.

About the initContainer, it can't be done solely at calico startup since this IP can come and go at any time and it needs calico to watch for interface changes and adjust the NAT rules.